### ISSUE TITLE
Single slash path (root) should not be considered in the "hierarchical" rule

### ISSUE THREAD

**Issue by efebeydogan01** [restful-ma/rest-ruler#54]
**Describe the bug**
#52 described a bug regarding a path that consists of a single slash ("/") and the trailing slash rule. A similar problem occurs with the rule `Forward slash separator (/) must be used to indicate a hierarchical relationship`, where a violation is reported if the path is just a forward slash. We should not perform the hierarchy check in this situation.

**To Reproduce**
1. To quickly reproduce the issue, run this command (while you are at the root of the repo): `java -jar build/libs/rest-ruler.jar -p src\test\java\cli\rule\tunnelingTests\InvalidOpenAPITrailingRule.json` (we are using one of the test files that are already available in the repo, although make sure you build the binary from the latest commit)
2. You will see that a violation is reported for the root path:
```
| Line No. | Line                 | Rule Violated                                                                           |
| -------- | -------------------- | --------------------------------------------------------------------------------------- |
...
| 121      | /                    | Forward slash separator (/) must be used to indicate a hierarchical relationship        |
...
```

**Expected behavior**
A violation should not be reported, as there are no nodes in the URI that could even be considered for a hierarchical analysis.
